wordpress-mobile/WordPress-iOS
# Migrate Pages List to Swift.

Migrates PageListViewController to Swift.  Since porting just a part of the code is not feasible, I'm describing below how this'll be taken care of.

**Porting and reviewing guidelines:**

First off I've set some porting rules, in order to make the work easier.  Please keep these in mind when reviewing the code:

- The ported code doesn't need to be better, just equivalent.  Improvements can be made after this initial porting step is complete.
- Additional comments won't be added.  Documenting methods is outside the scope of this particular PR.
- Make sure the code is 1:1 as much as possible (saving the occasional improvement).

**Code Reviews:**

I've split the review in sections (the `#pragma mark` sections from the original ObjC file), so that the reviewers can focus on smaller parts of the whole thing.

Please review @jleandroperez
- [x] Header File
- [x] Constants & Properties
- [x] Lifecycle Methods
- [x] Sync Methods
- [x] TableView Handler Delegate Methods

Please review @SergioEstevao  
- [x] Table View Handling
- [x] Post Actions
- [x] Filter related
- [x] Cell Delegate Methods

**Testing:**

Please make sure to run these tests both with WP.com and self-hosted.

Steps to find this VC in the app:

1. Go to My Sites (first icon with the WordPress logo in the tab bar)
2. Select a site
3. Tap "Blog Posts"... et voila!

Tests for @jleandroperez:

1. Make sure you open a blog without pages.
2. Make sure that even if you select any filter from the top, no pages are shown on screen.
3. Make sure that each section is showing an appropriate message at the center of the screen (you'll know when you read them!).


1. Make sure you open a blog without pages.
2. Tap the search icon.
3. Type in any word.
4. Make sure the text at the center of the screen shows an appropriate message for the filter you've selected.

Tests for @SergioEstevao 

Same as test 2 but with results.  Make sure the results are shown properly.

1. Add pages through the blog's web page.
2. Make sure at least 1 of them is published, 1 draft, 1 scheduled and 1 trashed.
3. Check in the app to make sure the pages are in the appropriate section.

1. Same as test 4 but once you have the pages...
2. Move them around or delete them using WPiOS, and make sure the changes are properly reflected in the web interface.


- [ ] Test 6:
1. Create a new page in WPiOS for each one of these: published, draft, scheduled.
2. Make sure they show up properly and that they're uploaded to your blog.

And any other test you can think of in that screen. :)
